src/peer-connection.spec.ts

```diff
@@ -247,17 +248,76 @@ describe('PeerConnection', () => {
     });
   });
 
+  describe('createOffer', () => {
+    let mockPc: MockedObjectDeep<RTCPeerConnectionStub>;
+    let createOfferSpy: jest.SpyInstance;
+    const callback = jest.fn();
+    let pc: PeerConnection;
+    const mockedReturnedOffer: RTCSessionDescriptionInit = {
+      sdp: 'blah',
+      type: 'offer',
+    };
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      mockPc = mocked(new RTCPeerConnectionStub(), true);
+      mockPc.createOffer.mockImplementation((description) => {
+        return new Promise((resolve, reject) => {
+          if (!description) {
+            reject(new Error());
+          } else {
+            resolve(mockedReturnedOffer);
+          }
+        });
+      });
+      mockCreateRTCPeerConnection.mockReturnValueOnce(mockPc as unknown as RTCPeerConnection);
+      pc = new PeerConnection();
+      createOfferSpy = jest.spyOn(pc, 'createOffer');
+      pc.on(PeerConnection.Events.CreateOfferOnSuccess, callback);
+    });
+
+    it('should emit event when createOffer called', async () => {
+      expect.hasAssertions();
+      const options: RTCOfferOptions = {
+        iceRestart: true,
+      };
+      const offer = await pc.createOffer(options);
+      expect(offer).toStrictEqual(mockedReturnedOffer);
+      expect(createOfferSpy).toHaveBeenCalledWith(options);
+      expect(callback).toHaveBeenCalledWith(mockedReturnedOffer);
+    });
+
+    it('should not emit event when createOffer failed', async () => {
+      expect.hasAssertions();
+      const offerPromise = pc.createOffer(null as unknown as RTCOfferOptions);
+      await expect(offerPromise).rejects.toThrow(Error);
+      expect(createOfferSpy).toHaveBeenCalledWith(null);
+      expect(callback).toHaveBeenCalledTimes(0);
+    });
+  });
+
   describe('setLocalDescription', () => {
-    let mockPc: RTCPeerConnectionStub;
+    let mockPc: MockedObjectDeep<RTCPeerConnectionStub>;
     let setLocalDescriptionSpy: jest.SpyInstance;
+    const callback = jest.fn();
     let pc: PeerConnection;
 
     beforeEach(() => {
       jest.clearAllMocks();
       mockPc = mocked(new RTCPeerConnectionStub(), true);
       mockCreateRTCPeerConnection.mockReturnValueOnce(mockPc as unknown as RTCPeerConnection);
+      mockPc.setLocalDescription.mockImplementation((description) => {
```

Comment:
`description` may not be the best param name for this, since you typically don't pass an SDP into a call to `createOffer`. I also think it may be confusing to throw if no param is provided, since technically `createOffer` can be called with no params (same with `setLocalDescription`).

Wondering if it might be cleaner to mock `createOffer` so it always succeeds here, and then overwrite the mock to fail only in the test cases where we do want it to fail?